### ISSUE TITLE
NPE when using arrayOfObjects with template

### ISSUE THREAD

**Issue by mikesir87** [soulwing/prospecto#16]
### Domain classes

```java
public class OptionCollection {
  private Set<Option> options = new HashSet<>();
  public Set<Option> getOptions() { return options; }
}

public class Option {
  private String label;
  public String getLabel() {  return label; }
}
```

### Templates

```java
  static ViewTemplate OPTION_TEMPLATE = ViewTemplateBuilderProducer
      .object(Option.class)
        .value("value")
        .build();

  static ViewTemplate TEMPLATE = ViewTemplateBuilderProducer
      .object(OptionCollection.class)
        .arrayOfObjects("options", OPTION_TEMPLATE)
        .build();
```

The template building then throws:

```
Caused by: java.lang.NullPointerException
	at org.soulwing.prospecto.runtime.builder.ValueNodeViewTemplateBuilder.onEnd(ValueNodeViewTemplateBuilder.java:112)
	at org.soulwing.prospecto.runtime.builder.AbstractViewTemplateBuilder.end(AbstractViewTemplateBuilder.java:424)
	at org.soulwing.prospecto.runtime.builder.ValueNodeViewTemplateBuilder.build(ValueNodeViewTemplateBuilder.java:117)
	at com.campstur.api.views.ViewMain.<clinit>(ViewMain.java:68)
	... 3 more
```

where line 68 is on the .build() for the TEMPLATE.

**Comment by ceharris** [comment]
Ok... I can reproduce this. Not sure what's going on just yet.

**Comment by ceharris** [comment]
Fixed by 7c16151
